docs/standard/microservices-architecture/multi-container-microservice-net-applications/implementing-api-gateways-with-ocelot.md

```diff
@@ -0,0 +1,579 @@
+author: CESARDELATORRE
+ms.author: wiwagn
+ms.date: 06/06/2018
+---
+# Implementing API Gateways with Ocelot
+
+The reference microservice application [eShopOnContainers](https://github.com/dotnet-architecture/eShopOnContainers) is using [Ocelot](https://github.com/ThreeMammals/Ocelot) because Ocelot is a simple and lightweight API Gateway that you can deploy anywhere along with your microservices/containers such as in any of the following environments used by eShopOnContainers.
+
+-	Docker host, in your local dev PC, on-premises or in the cloud
+-	Kubernetes cluster, on-premises or in managed cloud such as Azure Kubernetes Service (AKS)
+-	Service Fabric cluster, on-premises or in the cloud
+-	Service Fabric mesh, as PaaS/Serverless in Azure
+
+## Architect and design your API Gateways
+
+The following architecture diagram shows how API Gateways are implemented with Ocelot in eShopOnContainers.
+
+![eShopOnContainers architecture diagram showing client apps, microservices and the API Gateways in between](./media/image28.png)
+
+**Figure 8-27.** eShopOnContainers architecture with API Gateways
+
+That diagram shows how the whole application is deployed into a single Docker host or development PC with “Docker for Windows” or “Docker for Mac”. However, deploying into any orchestrator would be pretty similar but any container in the diagram could be scaled-out in the orchestrator. 
+
+In addition, the infrastructure assets such as databases, cache, and message brokers should be offloaded from the orchestrator and deployed into high available systems for infrastructure, like Azure SQL Database, Azure Cosmos DB, Azure Redis, Azure Service Bus, or any HA clustering solution on-premises.
+
+As you can also notice in the diagram, having several API Gateways allows multiple development teams to be autonomous (in this case Marketing features vs. Shopping features) when developing and deploying their microservices plus their own related API Gateways. 
+
+If you had a single monolithic API Gateway that would mean a single point to be updated by several development teams, which could couple all the microservices with a single part of the application.
+
+Going much further in the design, sometimes a fine-grained API Gateway can also be limited to a single business microservice depending on the chosen architecture. Having the API Gateway’s boundaries dictated by the business or domain will help you to get a better design. 
+
+For instance, fine granularity in the API Gateway tier can be especially useful for more advanced composite UI applications that are based on microservices, because the concept of a fine-grained API Gateway is similar to a UI composition service. 
+
+You can read further information about this topic in the section [Creating composite UI based on microservices](https://docs.microsoft.com/dotnet/standard/microservices-architecture/architect-microservice-container-applications/microservice-based-composite-ui-shape-layout).
```

Comment:
topic -> article

The boilerplate sentence for this situation is "For more information about X, see Y".
We also use relative paths for internal links so we can get links validated at build time.

Reply:
👍 - 
About the link, I'd prefer to keep the full path so it is consistent with the WORD/PDF version.